### ISSUE TITLE
Change  bibliographystyle in elsevier_article when using natbib

### ISSUE THREAD

**Issue by hongyuanjia** [rstudio/rticles#395]
### Why natbib but not csl?

This is because I want to get a clean TeX file together with the output PDF. Using csl will write all references into the TeX file, e.g.

```tex
\hypertarget{refs}{}
\begin{CSLReferences}{0}{0}
\leavevmode\hypertarget{ref-jia2021eplusr}{}%
\CSLLeftMargin{{[}1{]} }
\CSLRightInline{H. Jia, A. Chong, Eplusr: {A} framework for integrating building energy simulation and data-driven analytics, Energy and Buildings. 237 (2021) 110757. doi:\href{https://doi.org/10.1016/j.enbuild.2021.110757}{10.1016/j.enbuild.2021.110757}.}

\leavevmode\hypertarget{ref-crawley2001energyplus}{}%
\CSLLeftMargin{{[}2{]} }
\CSLRightInline{D.B. Crawley, L.K. Lawrie, F.C. Winkelmann, W.F. Buhl, Y.J. Huang, C.O. Pedersen, et al., {EnergyPlus}: Creating a new-generation building energy simulation program, Energy and Buildings. 33 (2001) 319--331. doi:\href{https://doi.org/10.1016/s0378-7788(00)00114-6}{10.1016/s0378-7788(00)00114-6}.}

\end{CSLReferences}
```

Currently `elsevier_article` hard-codes `\bibliographystyle{plainnat}`:

https://github.com/rstudio/rticles/blob/d93d50b39933b6ad91c245c95c008a5e3544716d/inst/rmarkdown/templates/elsevier/resources/template.tex#L58-L63

The default Pandoc LaTeX template exposes an option `biblio-style` to let the user to define the bibliography style. It is possible to implement this in the `elsevier_article`? Thanks.

https://github.com/jgm/pandoc-templates/blob/master/default.latex#L365-L374

```tex
$if(natbib)$
\usepackage[$natbiboptions$]{natbib}
\bibliographystyle{$if(biblio-style)$$biblio-style$$else$plainnat$endif$}
$endif$
$if(biblatex)$
\usepackage[$if(biblio-style)$style=$biblio-style$,$endif$$for(biblatexoptions)$$biblatexoptions$$sep$,$endfor$]{biblatex}
$for(bibliography)$
\addbibresource{$bibliography$}
$endfor$
$endif$
```

**Comment by cderv** [comment]
I think this has been closed by #402. You can now use `biblio-style` using dev version of **rticles**

Thanks for the suggestion !

**Comment by hongyuanjia** [comment]
Great! Thanks!
